3684/1/info_taker.rb

```diff
@@ -0,0 +1,53 @@
+require 'open-uri'
+require 'nokogiri'
+
+class InfoTaker
+  attr_reader :adress
+
+  def initialize(adress)
+    @adress = adress
+    @document = open_doc
+    @documnet_for_used_by = open_doc("#{adress}/network/dependents")
+    @info = {}
+  end
+
+  def take_info
+    @info.store(:used_by, used_by)
+    parse_adresses
+  end
+
+  private
+
+  def open_doc(adress = @adress)
+    Nokogiri::HTML(::Kernel.open(adress))
+  end
+
+  def parse_adresses
+    parse_htm
```

Comment:
move this method call to the `take_info` before  `parse_adresses` method call
`parse_htm` doesn't depend on `parse_adresses` and could be called independently